commit 075163bbb0f51174359947e1bce84f5edb23f21e
Author: Sujith Manoharan <c_manoha@qca.qualcomm.com>
Date:   Thu Jun 20 13:57:07 2013 +0530

    ath9k_htc: Handle IDLE state transition properly
    
    Make sure that a chip reset is done when IDLE is turned
    off - this fixes authentication timeouts.
    
    Cc: stable@vger.kernel.org
    Reported-by: Ignacy Gawedzki <i@lri.fr>
    Signed-off-by: Sujith Manoharan <c_manoha@qca.qualcomm.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/ath/ath9k/htc_drv_main.c b/drivers/net/wireless/ath/ath9k/htc_drv_main.c
index 0743a47cef8f..62f1b7636c92 100644
--- a/drivers/net/wireless/ath/ath9k/htc_drv_main.c
+++ b/drivers/net/wireless/ath/ath9k/htc_drv_main.c
@@ -1174,7 +1174,7 @@ static int ath9k_htc_config(struct ieee80211_hw *hw, u32 changed)
 		mutex_lock(&priv->htc_pm_lock);
 
 		priv->ps_idle = !!(conf->flags & IEEE80211_CONF_IDLE);
-		if (priv->ps_idle)
+		if (!priv->ps_idle)
 			chip_reset = true;
 
 		mutex_unlock(&priv->htc_pm_lock);